Development/NutriMon/src/main/java/chillingMonsters/Controllers/UserProfileController.java

```diff
@@ -0,0 +1,110 @@
+package chillingMonsters.Controllers;
+
+import chillingMonsters.DBConnect;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class UserProfileController extends NutriMonController {
+    UserProfileController() {
+        super("userProfile", "userID");
+    }
+
+    /**
+     * A function checks the email and password combination
+     *
+     * @param userEmail input user email
+     * @param password  input password
+     * @return true if email and password pair is found
+     */
+    public boolean checkCredentials(String userEmail, String password) {
+        Map<String, String> userProfile = new HashMap<>();
+        String queryString = "SELECT userEmail, `password` " +
```

Comment:
Why do we need to retrieve the password here, shouldn't we retrieve the userId and set it in the controller instead?

Reply:
We don't want to retrieve the userID before login. This function is called before login. UserID is retrieved in Landing page

Reply:
This function is literally called at the end of loginButton, and after that it prompts a successful login, why shouldn't it set the userID? Since every page after login needs a consistent user, retrieve userID on landing page is inconsistent